src/java.base/share/classes/sun/nio/fs/AbstractWatchKey.java

```diff
@@ -35,9 +36,12 @@
 abstract class AbstractWatchKey implements WatchKey {
 
     /**
-     * Maximum size of event list (in the future this may be tunable)
+     * Maximum size of event list
      */
-    static final int MAX_EVENT_LIST_SIZE    = 512;
+    @SuppressWarnings("removal")
+    static final int MAX_EVENT_LIST_SIZE =
+        AccessController.doPrivileged((PrivilegedAction<Integer>) () ->
+                Integer.getInteger("sun.nio.fs.maxWatchEvents", 512));
```

Comment:
If this system property is going to be used outside of the JDK testing then it will need to be renamed to a jdk.* property and we'll need to create a CSR.

Also the implementation should use GetPropertyAction.privilegedGetProperty.

Reply:
> If this system property is going to be used outside of the JDK testing then it will need to be renamed to a jdk.* property and we'll need to create a CSR.

I renamed it to `jdk.nio.file.maxWatchEvents`, is that what you would expect?

Is a CSR something I can create myself? I followed the linked docs but it looks like I can't to do much without an OpenJDK account.

> Also the implementation should use GetPropertyAction.privilegedGetProperty.

Done.